packages/grafana-data/src/types/datasource.ts

```diff
@@ -599,6 +599,7 @@ export interface DataSourceInstanceSettings<T extends DataSourceJsonData = DataS
   password?: string; // when access is direct, for some legacy datasources
   database?: string;
   isDefault?: boolean;
+  access?: string;
```

Comment:
would it make sense in your opinion to have `'direct' | 'server'` instead?
also in https://github.com/grafana/grafana/blob/32a551b4caae9ac6c6c4ba3d006d7096a9023a02/packages/grafana-data/src/types/datasource.ts#L568

also, I think this is always set (at least according to the types above), so what do you think about
```suggestion
  access: string;
```
?

Reply:
Yeah you are right! I got slightly uncertain as the url is also optional, but it is non-optional in  https://github.com/grafana/grafana/blob/32a551b4caae9ac6c6c4ba3d006d7096a9023a02/packages/grafana-data/src/types/datasource.ts#L569 .